cloud-run/cloudbuild.yaml

```diff
@@ -0,0 +1,65 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  - id: 'Build Container Image'
+    name: 'gcr.io/cloud-builders/docker:latest'
+    dir: 'app'
+    args: ['build', '-t', 'gcr.io/${PROJECT_ID}/${_SERVICE}', '.'] # Tag docker image with git commit sha
+  - id: 'Push Container Image'
+    name: 'gcr.io/cloud-builders/docker:latest'
+    args: ['push', 'gcr.io/${PROJECT_ID}/${_SERVICE}']
+  - id: 'Deploy to Cloud Run'
+    name: 'gcr.io/google.com/cloudsdktool/cloud-sdk'
+    entrypoint: gcloud
+    args: ['run', 'deploy', '${_SERVICE}', '--image', 'gcr.io/${PROJECT_ID}/${_SERVICE}', '--region', '${_REGION}']
```

Comment:
can we also make sure the Cloud Run is not unauthenticted

Reply:
This is actually an authenticated service by default.